proxy-node-shared/src/main/java/uk/gov/ida/notification/configuration/CredentialConfiguration.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.ida.notification.configuration;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import org.opensaml.security.credential.Credential;
+import org.opensaml.security.x509.BasicX509Credential;
+import org.opensaml.xmlsec.signature.support.SignatureConstants;
+import uk.gov.ida.notification.saml.SamlObjectSigner;
+
+import java.security.interfaces.ECPublicKey;
+
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.NAME,
+    property = "type"
+)
+@JsonSubTypes({
+    @JsonSubTypes.Type(value= KeyFileCredentialConfiguration.class, name="file"),
+    @JsonSubTypes.Type(value= CloudHsmCredentialConfiguration.class, name="cloudhsm")
+})
+public abstract class CredentialConfiguration {
+    protected BasicX509Credential credential;
+    protected String algorithm;
+
+    public BasicX509Credential getCredential() { return credential; }
+    public String getAlgorithm() { return algorithm; }
+
+    public void setCredential(BasicX509Credential credential) {
+        this.algorithm = SignatureConstants.ALGO_ID_SIGNATURE_RSA_SHA256;
+        if (credential.getPublicKey() instanceof ECPublicKey) {
```

Comment:
Could this be a turnery

Reply:
Mebbe, but hopefully this makes it explicit that RSA is the default.